### PR TITLE
[P02-H02] Restrict liquidations by min collateralization ratio as well

### PR DESCRIPTION
**Pull request by ptare**
A liquidator can now specify a min and a max for collateralization ratio
of the position they are liquidating.

The idea is to avoid a delayed liquidation ending up liquidating a
position in haircut territory. E.g., if a liquidation takes a long time
to process and the underlying price moves significantly in that time.

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

**Comment by ptare** [comment]
I'm still updating the tests, but I wanted to see if we actually want to make this change or not. Thoughts? @chrismaree @nicholaspai @mrice32

File: core/contracts/financial-templates/implementation/Liquidatable.sol [hunk 1]
```diff
@@ -253,8 +253,16 @@ contract Liquidatable is PricelessPositionManager {
             })
         );
 
+        // Scoping to get rid of a stack too deep error.
         // Adjust the sponsor's remaining position.
-        _reduceSponsorPosition(sponsor, tokensLiquidated, lockedCollateral, withdrawalAmountToRemove);
+        {
+            // Part of the withdrawal request is also removed. Ideally:
+            // liquidatedCollateral + withdrawalAmountToRemove = lockedCollateral.
+            FixedPoint.Unsigned memory withdrawalAmountToRemove = positionToLiquidate.withdrawalRequestAmount.mul(
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/Liquidatable.sol:261]: Why is this added?

Comment (ptare) [reply core/contracts/financial-templates/implementation/Liquidatable.sol:261]: It's moved from above.

Comment (nicholaspai) [reply core/contracts/financial-templates/implementation/Liquidatable.sol:261]: got it. Wonder why the stack too deep error occurs with these changes

Comment (ptare) [reply core/contracts/financial-templates/implementation/Liquidatable.sol:261]: Because I added new local variable: the function parameter. Hate Solidity for these problems.

Comment (mrice32) [reply core/contracts/financial-templates/implementation/Liquidatable.sol:261]: Sucks :/
